internal/components/staticregistry/provide.go

```diff
@@ -0,0 +1,16 @@
+package staticregistry
+
+import (
+	"github.com/grafana/grafana/internal/components"
+	"github.com/grafana/grafana/internal/components/datasource"
+)
+
+// ProvideRegistry provides a simple static Registry.
+// Coremodels have to be manually added to this Registry.
+func ProvideRegistry(
+	datasourceModel *datasource.Coremodel,
+) (*components.Registry, error) {
+	return components.NewRegistry(
+		datasourceModel,
+	)
+}
```

Comment:
i don't see how enterprise could extend this, so not possible to do :/

Reply:
Yup, so this is the biggest problem at the moment, but I think we should postpone it to the next phase.

The problem with dynamic registration is the fact that we don't have an easy way to inject store implementations to components, which is the next thing on my list to figure out.